portal-impl/src/com/liferay/portlet/announcements/service/impl/AnnouncementsEntryLocalServiceImpl.java

```diff
@@ -583,6 +584,14 @@ protected void validate(
 			throw new EntryTitleException();
 		}
 
+		int titleMaxLength = ModelHintsUtil.getMaxLength(
+			AnnouncementsEntry.class.getName(), "title");
+
+		if (title.length() > titleMaxLength) {
+			throw new EntryTitleException(
+				"Title has more than " + titleMaxLength + " characters");
+		}
+
```

Comment:
@binhtran92 
Please consider using the https://help.liferay.com/hc/en-us/articles/360018179851-Using-the-AlloyUI-Validator-Tag on the Front-end jsp the title input field. Since the JSP files is just in side the Announcement module. I think It is possible to do that.
Thank you.